rotkehlchen/tests/api/test_exchanges.py

```diff
@@ -1245,3 +1245,99 @@ def test_binance_query_pairs(rotkehlchen_api_server_with_exchanges):
     assert 'FTTBNB' not in result
     if ci_run is False:
         assert binance_pairs_num > binanceus_pairs_num
+
+
+@pytest.mark.parametrize('added_exchanges', [(Location.BINANCE,)])
+def test_get_binance_savings_balance(rotkehlchen_api_server_with_exchanges):
+    """Check that querying the binance savings balance endpoint returns the expected response."""
+    async_query = random.choice([True, False])
+
+    # check for errors
+    response = requests.get(
+        api_url_for(
+            rotkehlchen_api_server_with_exchanges,
+            'binancesavingsresource',
+            location=Location.KRAKEN.name.lower(),
+        ),
+    )
+    assert_error_response(response, 'is not one of binance,binanceus')
+
+    interests_history_call_count = 0
+    lending_balances_call_count = 0
+
+    def mock_api_query(api_type, method, options):  # pylint: disable=unused-argument
+        nonlocal interests_history_call_count
+        nonlocal lending_balances_call_count
+
+        if 'lending/union/account' in method:
+            if lending_balances_call_count >= 1:
```

Comment:
have you checked that the 3 types of lending types are correctly tracked?

Reply:
yeah

Reply:
can you also assert that all the 3 valid lending types were queried? Even you could fake a different asset per response to validate it and would be better